google/cloud/storage/bucket_metadata_test.cc

```diff
@@ -958,27 +977,31 @@ TEST(BucketMetadataPatchBuilder, ResetDefaultAcl) {
 
 TEST(BucketMetadataPatchBuilder, SetIamConfiguration) {
   BucketMetadataPatchBuilder builder;
-  std::string expected =
-      "projects/test-project-name/locations/us-central1/keyRings/"
-      "test-keyring-name/cryptoKeys/test-key-name";
-  builder.SetEncryption(BucketEncryption{expected});
+  BucketIamConfiguration config;
+  config.public_access_prevention = "enforced";
+  config.uniform_bucket_level_access = UniformBucketLevelAccess{true, {}};
+  builder.SetIamConfiguration(config);
 
   auto actual = builder.BuildPatch();
   auto json = nlohmann::json::parse(actual);
-  ASSERT_EQ(1U, json.count("encryption")) << json;
-  ASSERT_TRUE(json["encryption"].is_object()) << json;
-  EXPECT_EQ(expected, json["encryption"].value("defaultKmsKeyName", ""))
-      << json;
+  ASSERT_EQ(1U, json.count("iamConfiguration")) << json;
+  ASSERT_TRUE(json["iamConfiguration"].is_object()) << json;
+  auto const expected = nlohmann::json{
+      {"publicAccessPrevention", "enforced"},
+      {"uniformBucketLevelAccess", {{"enabled", true}}},
+      {"bucketPolicyOnly", {{"enabled", true}}},
+  };
+  EXPECT_EQ(json["iamConfiguration"], expected);
 }
 
 TEST(BucketMetadataPatchBuilder, ResetIamConfiguration) {
   BucketMetadataPatchBuilder builder;
-  builder.ResetEncryption();
+  builder.ResetIamConfiguration();
 
   auto actual = builder.BuildPatch();
   auto json = nlohmann::json::parse(actual);
-  ASSERT_EQ(1U, json.count("encryption")) << json;
-  ASSERT_TRUE(json["encryption"].is_null()) << json;
+  ASSERT_EQ(1U, json.count("iamConfiguration")) << json;
+  ASSERT_TRUE(json["iamConfiguration"].is_null()) << json;
 }
 
```

Comment:
I don't see an integration test for this feature (aside from the sample test), am I missing it?

Reply:
Not missing it, but frankly we do not have an integration tests for each field. Unit tests + a generic integration test is how we roll.

Reply:
Got it, if that's how it works in this library then it's fine with me.

Reply:
Added the integration tests, had to disable it because it does not pass against production with my personal project 🤷